theia-extensions/viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -11,25 +11,33 @@ export class TraceServerServiceImpl implements TraceServerConfigService {
     private server?: ChildProcess;
     private port?: number;
 
-    async startTraceServer(path?: string, port?: number): Promise<string> {
+    async startTraceServer(path?: string, cliArgs?: string): Promise<string> {
         if (path === undefined) {
             throw new Error('no Trace Server path specified');
-        } else if (port === undefined) {
+        } else if (cliArgs === undefined) {
```

Comment:
The cli arguments should be optional. For example, it should be possible to start the Trace Compass server without any arguments. This will start the TC server with default port and workspace.

Reply:
Something I learned recently with JSON-RPC services is that if the method parameters are positional (just like they are now) then `undefined` values are transformed into `null` values.

Two possible fixes:
- Just write `if (!path) {` so that either `undefined` or `null` work the same way.
- Replace the positional arguments with an `options` object containing the potentially undefined fields as those will properly carry the `undefined` values...

I actually changed to the latter in a different PR. This is a limitation of JSON-RPC :/

Reply:
Replace the positional arguments with an options object as done in PR #594